crates/astria-composer/src/executor/mod.rs

```diff
@@ -276,7 +287,101 @@ impl Executor {
+            }
+        };
+
+        let mut total_bundles_drained_successfully = 0;
+        let shutdown_logic = async {
+            info!("setting executor connection status to disconnected");
+            self.status
+                .send_modify(|status| status.is_connected = false);
+
+            info!("draining sequence actions from the executor receiver channel");
+            // drain the receiver channel
+            while let Ok(seq_action) = self.serialized_rollup_transactions.try_recv() {
+                let rollup_id = seq_action.rollup_id;
+                if let Err(e) = bundle_factory.try_push(seq_action) {
+                    warn!(
+                        rollup_id = %rollup_id,
+                        error = &e as &StdError,
+                        "failed to bundle sequence action: too large. sequence action is dropped."
+                    );
+                }
+            }
+
+            info!("draining remaining bundles from bundle factory");
+            // when shutting down, drain all the remaining bundles and submit to the sequencer
+            // to avoid any bundle loss.
+            loop {
+                let bundle = bundle_factory.pop_now();
+                if bundle.is_empty() {
+                    break;
+                }
+                match self.submit_bundle(nonce, bundle.clone()).await {
+                    Ok(new_nonce) => {
+                        debug!(
+                            bundle = %telemetry::display::json(&SizedBundleReport(&bundle)),
+                            "drained bundle successfully; new nonce: {}", new_nonce
+                        );
+                        nonce = new_nonce;
+                        total_bundles_drained_successfully += 1;
+                    }
+                    Err(e) => {
+                        error!(bundle =  %telemetry::display::json(&SizedBundleReport(&bundle)),
+                            %e, "failed submitting bundle to sequencer during shutdown; \
+                                aborting shutdown");
+                        // if we can't submit a bundle after multiple retries, we can abort
+                        // the shutdown process
+                        return Err(e);
+                    }
+                }
+            }
+
+            Ok(())
+        };
+
+        match tokio::time::timeout(Duration::from_secs(EXECUTOR_SHUTDOWN_TIME), shutdown_logic)
+            .await
+        {
+            Ok(res) => {
+                if let Err(e) = res {
+                    error!(%e, "executor shutdown tasks aborted");
+                } else {
+                    info!("executor shutdown tasks completed successfully");
+                }
+            }
+            Err(err) => {
+                error!(%err, "executor shutdown tasks failed to complete in time");
+            }
+        }
+
+        info!(
+            "{} bundles drained successfully during shutdown",
+            total_bundles_drained_successfully
```

Comment:
This is a pretty confusing message if there were no bundles to be submitted ever:

Maybe:
```rust
info!(total, "bundles submitted during shutdown");
```